### PR TITLE
Modify the createcheckpoint interface of storage

### PR DESCRIPTION
**Pull request by monadbobo**
depend https://github.com/vesoft-inc/nebula-common/pull/368

### CODE REVIEW

File: src/meta/processors/admin/SnapShot.cpp [hunk 1]
```diff
@@ -13,22 +13,38 @@
...
             if (!status.ok()) {
                 return cpp2::ErrorCode::E_RPC_FAILURE;
             }
-            info[spaceHosts.first].emplace_back(
-                apache::thrift::FRAGILE, host, status.value());
+            auto infoPair = status.value();
+            auto it = info.find(spaceHosts.first);
+            if (it != info.cend()) {
+                it->second.second.emplace_back(
+                    apache::thrift::FRAGILE, host, std::move(infoPair.first));
```
Comment (bright-starry-sky) [inline src/meta/processors/admin/SnapShot.cpp:40]: A suggestion, don't use `apache::thrift::FRAGILE`, there will be version compatibility issues in the future.

File: src/meta/processors/admin/SnapShot.cpp [hunk 2]
```diff
@@ -13,22 +13,38 @@
...
+            } else {
```
Comment (bright-starry-sky) [inline src/meta/processors/admin/SnapShot.cpp:41]: Please add some comments for map of `info`?
what about the map as below ?
```
std::unordered_map<
        GraphSpaceID,
        std::unordered_map<std::vector<cpp2::CheckpointInfo>, nebula::cpp2::PartitionBackupInfo>>
```

File: src/meta/processors/admin/SnapShot.cpp [hunk 3]
```diff
@@ -13,22 +13,38 @@
 
 namespace nebula {
 namespace meta {
-ErrorOr<cpp2::ErrorCode, std::unordered_map<GraphSpaceID, std::vector<cpp2::CheckpointInfo>>>
+ErrorOr<cpp2::ErrorCode,
```
Comment (liuyu85cn) [inline src/meta/processors/admin/SnapShot.cpp:16]: how about add some using to shorten this return type?

**Review by liuyu85cn — APPROVED** [review]
LGTM

**Comment by bright-starry-sky** [comment]
LGTM

**Review by bright-starry-sky — APPROVED** [review]

File: src/kvstore/KVStore.h [hunk 4]
```diff
@@ -192,8 +192,8 @@ class KVStore {
 
     virtual ResultCode flush(GraphSpaceID spaceId) = 0;
 
-    virtual ErrorOr<ResultCode, std::string> createCheckpoint(GraphSpaceID spaceId,
-                                                              const std::string& name) = 0;
+    virtual ErrorOr<ResultCode, std::pair<std::string, nebula::cpp2::PartitionBackupInfo>>
```
Comment (critical27) [inline src/kvstore/KVStore.h:195]: We need to change the data structure of resp. If a space is mounted on multiple disks, only one path is not sufficient.
Something like `map<path, PartitionBackupInfo>` is ok.

Comment (critical27) [reply src/kvstore/KVStore.h:195]: Talked offline, will fix in #377

File: src/kvstore/raftex/RaftPart.cpp [hunk 5]
```diff
@@ -1891,6 +1891,10 @@ std::set<HostAddr> RaftPart::listeners() const {
     return listeners_;
 }
 
+std::pair<LogID, TermID> RaftPart::lastLogInfo() const {
+    return std::make_pair(wal_->lastLogId(), wal_->lastLogTerm());
```
Comment (critical27) [inline src/kvstore/raftex/RaftPart.cpp:1895]: Better grab the lock here.

File: src/kvstore/NebulaStore.cpp [hunk 6]
```diff
@@ -924,9 +925,19 @@ ErrorOr<ResultCode, std::string> NebulaStore::createCheckpoint(GraphSpaceID spac
             if (!p->linkCurrentWAL(walPath.data())) {
                 return ResultCode::ERR_CHECKPOINT_ERROR;
             }
+
+            if (p->isLeader()) {
```
Comment (critical27) [inline src/kvstore/NebulaStore.cpp:929]: Why only leader report wal info?

**Review by critical27 — APPROVED** [review]

**Pull request merged**